build fails under node v0.11.13

Building geoip fails with node v.0.11.13.

$ node --version
v0.11.13
$ npm install geoip
...
In file included from ../src/global.h:10:0,
                 from ../src/global.cc:7:
/home/wtf/a/node_modules/geoip/node_modules/nan/nan.h: In function ‘v8::Handlev8::Value NanError(c
onst char*)’:
/home/wtf/a/node_modules/geoip/node_modules/nan/nan.h:319:38: error: ‘New’ is not a member of ‘v8::S
tring’
 # define _NAN_ERROR(fun, errmsg) fun(v8::String::New(errmsg))
                                      ^
/home/wtf/a/node_modules/geoip/node_modules/nan/nan.h:339:13: note: in expansion of macro ‘_NAN_ERRO
R’
     return  _NAN_ERROR(v8::Exception::Error, errmsg);
             ^

--- a/binding.gyp
+++ b/binding.gyp
@@ -22,7 +22,7 @@
       'include_dirs': [
           'src',
           'deps/geoip-api-c-1.6.0/libGeoIP',
-          "<!(node -p -e \"require('path').dirname(require.resolve('nan'))\")"
+          "<!(node -e \"require('nan')\")"
       ],
       'dependencies': [
         'deps/geoip-api-c-1.6.0/libgeoip.gypi:libgeoip'

--- a/package.json
+++ b/package.json
@@ -18,7 +18,7 @@
     "James Taylor <jt@gosquared.com>",
     "Christopher Holt <cwholt@gmail.com>",
     "Nicolas Fouché <nicolas.fouche@gmail.com>",
-    "King Koopa <bbyholm@abo.fi>",
+    "Benjamin Byholm <bbyholm@abo.fi>",
     "Bernhard Valenti <bernhard@kargo.com>",
     "Pawel Maciejewski <fragphace@gmail.com>",
     "Ilya Furman <smashlong@gmail.com>",
@@ -26,7 +26,7 @@
     "Nicolas Fouché <nicolas.fouche@gmail.com>"
   ],
   "dependencies": {
-    "nan": "~0.8.0",
+    "nan": "~1.0.0",
     "bindings": "~1.1.1",
     "debug": "~0.7.4",
     "is-object": "~0.1.2"

--- a/src/city.cc
+++ b/src/city.cc
@@ -21,14 +21,14 @@ Persistent<FunctionTemplate> City::constructor_template;
 void City::Init(Handle<Object> exports) {
   NanScope();
 
-  Local<FunctionTemplate> tpl = NanNewLocal<FunctionTemplate>(FunctionTemplate::New(New));
-  NanAssignPersistent(FunctionTemplate, constructor_template, tpl);
+  Local<FunctionTemplate> tpl = NanNew<FunctionTemplate>(New);
+  NanAssignPersistent(constructor_template, tpl);
   tpl->InstanceTemplate()->SetInternalFieldCount(1);
-  tpl->SetClassName(String::NewSymbol("City"));
+  tpl->SetClassName(NanSymbol("City"));
 
-  tpl->PrototypeTemplate()->Set(String::NewSymbol("lookupSync"),
-      FunctionTemplate::New(lookupSync)->GetFunction());
-  exports->Set(String::NewSymbol("City"), tpl->GetFunction());
+  tpl->PrototypeTemplate()->Set(NanSymbol("lookupSync"),
+      NanNew<FunctionTemplate>(lookupSync)->GetFunction());
+  exports->Set(NanSymbol("City"), tpl->GetFunction());
 }
 
 NAN_METHOD(City::New) {
@@ -62,13 +62,12 @@ NAN_METHOD(City::lookupSync) {
 
   City *c = ObjectWrap::Unwrap<City>(args.This());
 
-  Local<Object> data = NanNewLocal<Object>(Object::New());
-  Local<String> host_str = NanNewLocal<String>(args[0]->ToString());
-
-  char host_cstr[host_str->Length() + 1];
-  NanFromV8String(args[0].As<Object>(), Nan::ASCII,
-          NULL, host_cstr, host_str->Length() + 1,
-          v8::String::HINT_MANY_WRITES_EXPECTED);
+  Local<Object> data = NanNew<Object>();
+  Local<String> host_str = args[0]->ToString();
+  size_t size = host_str->Length() + 1;
+  char host_cstr[size];
+  size_t bc;
+  NanCString(args[0], &bc, host_cstr, size);
 
   uint32_t ipnum = _GeoIP_lookupaddress(host_cstr);
 
@@ -76,74 +75,74 @@ NAN_METHOD(City::lookupSync) {
   //printf("Ipnum is %d.", ipnum);
 
   if (ipnum == 0) {
-    NanReturnValue(Null());
+    NanReturnValue(NanNull());
   }
 
   GeoIPRecord *record = GeoIP_record_by_ipnum(c->db, ipnum);
 
   if (!record) {
-    NanReturnValue(Null());
+    NanReturnValue(NanNull());
   }
 
   if (record->country_code) {
-    data->Set(String::NewSymbol("country_code"), String::New(record->country_code));
+    data->Set(NanSymbol("country_code"), NanNew<String>(record->country_code));
   }
 
   if (record->country_code3) {
-    data->Set(String::NewSymbol("country_code3"), String::New(record->country_code3));
+    data->Set(NanSymbol("country_code3"), NanNew<String>(record->country_code3));
   }
 
   if (record->country_name) {
-    data->Set(String::NewSymbol("country_name"), String::New(record->country_name));
+    data->Set(NanSymbol("country_name"), NanNew<String>(record->country_name));
   }
 
   if (record->region) {
-    data->Set(String::NewSymbol("region"), String::New(record->region));
+    data->Set(NanSymbol("region"), NanNew<String>(record->region));
   }
 
   if (record->city) {
     char *name = _GeoIP_iso_8859_1__utf8(record->city);
 
     if (name) {
-      data->Set(String::NewSymbol("city"), String::New(name));
+      data->Set(NanSymbol("city"), NanNew<String>(name));
     }
 
     free(name);
   }
 
   if (record->postal_code) {
-    data->Set(String::NewSymbol("postal_code"), String::New(record->postal_code));
+    data->Set(NanSymbol("postal_code"), NanNew<String>(record->postal_code));
   }
 
   if (record->latitude >= -90 && record->latitude <= 90) {
-    data->Set(String::NewSymbol("latitude"), Number::New(record->latitude));
+    data->Set(NanSymbol("latitude"), NanNew<Number>(record->latitude));
   }
 
   if (record->longitude >= -180 && record->longitude <= 180) {
-    data->Set(String::NewSymbol("longitude"), Number::New(record->longitude));
+    data->Set(NanSymbol("longitude"), NanNew<Number>(record->longitude));
   }
 
   if (record->metro_code) {
-    data->Set(String::NewSymbol("metro_code"), Number::New(record->metro_code));
+    data->Set(NanSymbol("metro_code"), NanNew<Number>(record->metro_code));
   }
 
   if (record->dma_code) {
-    data->Set(String::NewSymbol("dma_code"), Number::New(record->dma_code));
+    data->Set(NanSymbol("dma_code"), NanNew<Number>(record->dma_code));
   }
 
   if (record->area_code) {
-    data->Set(String::NewSymbol("area_code"), Number::New(record->area_code));
+    data->Set(NanSymbol("area_code"), NanNew<Number>(record->area_code));
   }
 
   if (record->continent_code) {
-    data->Set(String::NewSymbol("continent_code"), String::New(record->continent_code));
+    data->Set(NanSymbol("continent_code"), NanNew<String>(record->continent_code));
   }
 
   if (record->country_code && record->region) {
     const char *time_zone = GeoIP_time_zone_by_country_and_region(record->country_code, record->region);
 
     if(time_zone) {
-      data->Set(String::NewSymbol("time_zone"), String::New(time_zone));
+      data->Set(NanSymbol("time_zone"), NanNew<String>(time_zone));
     }
   }
 

--- a/src/city6.cc
+++ b/src/city6.cc
@@ -23,14 +23,14 @@ Persistent<FunctionTemplate> City6::constructor_template;
 void City6::Init(Handle<Object> exports) {
   NanScope();
 
-  Local<FunctionTemplate> tpl = NanNewLocal<FunctionTemplate>(FunctionTemplate::New(New));
-  NanAssignPersistent(FunctionTemplate, constructor_template, tpl);
+  Local<FunctionTemplate> tpl = NanNew<FunctionTemplate>(New);
+  NanAssignPersistent(constructor_template, tpl);
   tpl->InstanceTemplate()->SetInternalFieldCount(1);
-  tpl->SetClassName(String::NewSymbol("City6"));
+  tpl->SetClassName(NanSymbol("City6"));
 
-  tpl->PrototypeTemplate()->Set(String::NewSymbol("lookupSync"),
-      FunctionTemplate::New(lookupSync)->GetFunction());
-  exports->Set(String::NewSymbol("City6"), tpl->GetFunction());
+  tpl->PrototypeTemplate()->Set(NanSymbol("lookupSync"),
+      NanNew<FunctionTemplate>(lookupSync)->GetFunction());
+  exports->Set(NanSymbol("City6"), tpl->GetFunction());
 }
 
 NAN_METHOD(City6::New) {
@@ -62,83 +62,85 @@ NAN_METHOD(City6::New) {
 NAN_METHOD(City6::lookupSync) {
   NanScope();
 
-  Local<Object> data = NanNewLocal<Object>(Object::New());
-  Local<String> host_str = NanNewLocal<String>(args[0]->ToString());
-  char host_cstr[host_str->Length() + 1];
-  NanFromV8String(args[0].As<Object>(), Nan::ASCII, NULL, host_cstr, host_str->Length() + 1, v8::String::HINT_MANY_WRITES_EXPECTED);
+  Local<Object> data = NanNew<Object>();
+  Local<String> host_str = args[0]->ToString();
+  size_t size = host_str->Length() + 1;
+  char host_cstr[size];
+  size_t bc;
+  NanCString(args[0], &bc, host_cstr, size);
   City6 *c = ObjectWrap::Unwrap<City6>(args.This());
 
   geoipv6_t ipnum_v6 = _GeoIP_lookupaddress_v6(host_cstr);
 
   if (__GEOIP_V6_IS_NULL(ipnum_v6)) {
-    NanReturnValue(Null());
+    NanReturnValue(NanNull());
   }
 
   GeoIPRecord *record = GeoIP_record_by_ipnum_v6(c->db, ipnum_v6);
 
   if (!record) {
-    NanReturnValue(Null());
+    NanReturnValue(NanNull());
   }
 
   if (record->country_code) {
-    data->Set(String::NewSymbol("country_code"), String::New(record->country_code));
+    data->Set(NanSymbol("country_code"), NanNew<String>(record->country_code));
   }
 
   if (record->country_code3) {
-    data->Set(String::NewSymbol("country_code3"), String::New(record->country_code3));
+    data->Set(NanSymbol("country_code3"), NanNew<String>(record->country_code3));
   }
 
   if (record->country_name) {
-    data->Set(String::NewSymbol("country_name"), String::New(record->country_name));
+    data->Set(NanSymbol("country_name"), NanNew<String>(record->country_name));
   }
 
   if (record->region) {
-    data->Set(String::NewSymbol("region"), String::New(record->region));
+    data->Set(NanSymbol("region"), NanNew<String>(record->region));
   }
 
   if (record->city) {
     char *name = _GeoIP_iso_8859_1__utf8(record->city);
 
     if (name) {
-      data->Set(String::NewSymbol("city"), String::New(name));
+      data->Set(NanSymbol("city"), NanNew<String>(name));
     }
 
     free(name);
   }
 
   if (record->postal_code) {
-    data->Set(String::NewSymbol("postal_code"), String::New(record->postal_code));
+    data->Set(NanSymbol("postal_code"), NanNew<String>(record->postal_code));
   }
 
   if (record->latitude >= -90 && record->latitude <= 90) {
-    data->Set(String::NewSymbol("latitude"), Number::New(record->latitude));
+    data->Set(NanSymbol("latitude"), NanNew<Number>(record->latitude));
   }
 
   if (record->longitude >= -180 && record->longitude <= 180) {
-    data->Set(String::NewSymbol("longitude"), Number::New(record->longitude));
+    data->Set(NanSymbol("longitude"), NanNew<Number>(record->longitude));
   }
 
   if (record->metro_code) {
-    data->Set(String::NewSymbol("metro_code"), Number::New(record->metro_code));
+    data->Set(NanSymbol("metro_code"), NanNew<Number>(record->metro_code));
   }
 
   if (record->dma_code) {
-    data->Set(String::NewSymbol("dma_code"), Number::New(record->dma_code));
+    data->Set(NanSymbol("dma_code"), NanNew<Number>(record->dma_code));
   }
 
   if (record->area_code) {
-    data->Set(String::NewSymbol("area_code"), Number::New(record->area_code));
+    data->Set(NanSymbol("area_code"), NanNew<Number>(record->area_code));
   }
 
   if (record->continent_code) {
-    data->Set(String::NewSymbol("continent_code"), String::New(record->continent_code));
+    data->Set(NanSymbol("continent_code"), NanNew<String>(record->continent_code));
   }
 
   if (record->country_code && record->region) {
     const char *time_zone = GeoIP_time_zone_by_country_and_region(record->country_code, record->region);
 
     if(time_zone) {
-      data->Set(String::NewSymbol("time_zone"), String::New(time_zone));
+      data->Set(NanSymbol("time_zone"), NanNew<String>(time_zone));
     }
   }
 

--- a/src/country.cc
+++ b/src/country.cc
@@ -22,14 +22,14 @@ Persistent<FunctionTemplate> Country::constructor_template;
 void Country::Init(Handle<Object> exports) {
   NanScope();
 
-  Local<FunctionTemplate> tpl = NanNewLocal<FunctionTemplate>(FunctionTemplate::New(New));
-  NanAssignPersistent(FunctionTemplate, constructor_template, tpl);
+  Local<FunctionTemplate> tpl = NanNew<FunctionTemplate>(New);
+  NanAssignPersistent(constructor_template, tpl);
   tpl->InstanceTemplate()->SetInternalFieldCount(1);
-  tpl->SetClassName(String::NewSymbol("Country"));
+  tpl->SetClassName(NanSymbol("Country"));
 
-  tpl->PrototypeTemplate()->Set(String::NewSymbol("lookupSync"),
-      FunctionTemplate::New(lookupSync)->GetFunction());
-  exports->Set(String::NewSymbol("Country"), tpl->GetFunction()); 
+  tpl->PrototypeTemplate()->Set(NanSymbol("lookupSync"),
+      NanNew<FunctionTemplate>(lookupSync)->GetFunction());
+  exports->Set(NanSymbol("Country"), tpl->GetFunction());
 }
 
 
@@ -64,24 +64,28 @@ NAN_METHOD(Country::lookupSync) {
 
   Country *c = ObjectWrap::Unwrap<Country>(args.This());
 
-  Local<Object> data = NanNewLocal<Object>(Object::New());
-  Local<String> host_str = NanNewLocal<String>(args[0]->ToString());
-  char host_cstr[host_str->Length() + 1];
-  NanFromV8String(args[0].As<Object>(), Nan::ASCII, NULL, host_cstr, host_str->Length() + 1, v8::String::HINT_MANY_WRITES_EXPECTED);
+  Local<Object> data = NanNew<Object>();
+  Local<String> host_str = args[0]->ToString();
+  size_t size = host_str->Length() + 1;
+  char host_cstr[size];
+  size_t bc;
+  NanCString(args[0], &bc, host_cstr, size);
+
   uint32_t ipnum = _GeoIP_lookupaddress(host_cstr);
+
   if (ipnum <= 0) {
-    NanReturnValue(Null());
+    NanReturnValue(NanNull());
   } else {
     int country_id = GeoIP_id_by_ipnum(c->db, ipnum);
     if (country_id == 0) {
-      NanReturnValue(Null());
+      NanReturnValue(NanNull());
     } else {
       char *name = _GeoIP_iso_8859_1__utf8(GeoIP_country_name[country_id]);
 
-      data->Set(String::NewSymbol("country_name"), String::New(name));
-      data->Set(String::NewSymbol("country_code"), String::New(GeoIP_country_code[country_id]));
-      data->Set(String::NewSymbol("country_code3"), String::New(GeoIP_country_code3[country_id]));
-      data->Set(String::NewSymbol("continent_code"), String::New(GeoIP_country_continent[country_id]));
+      data->Set(NanSymbol("country_name"), NanNew<String>(name));
+      data->Set(NanSymbol("country_code"), NanNew<String>(GeoIP_country_code[country_id]));
+      data->Set(NanSymbol("country_code3"), NanNew<String>(GeoIP_country_code3[country_id]));
+      data->Set(NanSymbol("continent_code"), NanNew<String>(GeoIP_country_continent[country_id]));
 
       free(name);
 

--- a/src/country6.cc
+++ b/src/country6.cc
@@ -22,14 +22,14 @@ Persistent<FunctionTemplate> Country6::constructor_template;
 void Country6::Init(Handle<Object> exports) {
   NanScope();
 
-  Local<FunctionTemplate> tpl = NanNewLocal<FunctionTemplate>(FunctionTemplate::New(New));
-  NanAssignPersistent(FunctionTemplate, constructor_template, tpl);
+  Local<FunctionTemplate> tpl = NanNew<FunctionTemplate>(New);
+  NanAssignPersistent(constructor_template, tpl);
   tpl->InstanceTemplate()->SetInternalFieldCount(1);
-  tpl->SetClassName(String::NewSymbol("Country6"));
+  tpl->SetClassName(NanSymbol("Country6"));
 
-  tpl->PrototypeTemplate()->Set(String::NewSymbol("lookupSync"),
-      FunctionTemplate::New(lookupSync)->GetFunction());
-  exports->Set(String::NewSymbol("Country6"), tpl->GetFunction()); 
+  tpl->PrototypeTemplate()->Set(NanSymbol("lookupSync"),
+      NanNew<FunctionTemplate>(lookupSync)->GetFunction());
+  exports->Set(NanSymbol("Country6"), tpl->GetFunction());
 }
 
 NAN_METHOD(Country6::New) {
@@ -68,26 +68,29 @@ NAN_METHOD(Country6::lookupSync) {
     return NanThrowError("Error: Database is not country ipv6 edition");
   }
 
-  Local<Object> data = NanNewLocal<Object>(Object::New());
-  Local<String> host_str = NanNewLocal<String>(args[0]->ToString());
-  char host_cstr[host_str->Length() + 1];
-  NanFromV8String(args[0].As<Object>(), Nan::ASCII, NULL, host_cstr, host_str->Length() + 1, v8::String::HINT_MANY_WRITES_EXPECTED);
+  Local<Object> data = NanNew<Object>();
+  Local<String> host_str = args[0]->ToString();
+  size_t size = host_str->Length() + 1;
+  char host_cstr[size];
+  size_t bc;
+
+  NanCString(args[0], &bc, host_cstr, size);
 
   geoipv6_t ipnum_v6 = _GeoIP_lookupaddress_v6(host_cstr);
 
   if (__GEOIP_V6_IS_NULL(ipnum_v6)) {
-    NanReturnValue(Null());
+    NanReturnValue(NanNull());
   } else {
     int country_id = GeoIP_id_by_ipnum_v6(c->db, ipnum_v6);
     if (country_id == 0) {
-      NanReturnValue(Null());
+      NanReturnValue(NanNull());
     } else {
       char *name = _GeoIP_iso_8859_1__utf8(GeoIP_country_name[country_id]);
 
-      data->Set(String::NewSymbol("country_name"), String::New(name));
-      data->Set(String::NewSymbol("country_code"), String::New(GeoIP_country_code[country_id]));
-      data->Set(String::NewSymbol("country_code3"), String::New(GeoIP_country_code3[country_id]));
-      data->Set(String::NewSymbol("continent_code"), String::New(GeoIP_country_continent[country_id]));
+      data->Set(NanSymbol("country_name"), NanNew<String>(name));
+      data->Set(NanSymbol("country_code"), NanNew<String>(GeoIP_country_code[country_id]));
+      data->Set(NanSymbol("country_code3"), NanNew<String>(GeoIP_country_code3[country_id]));
+      data->Set(NanSymbol("continent_code"), NanNew<String>(GeoIP_country_continent[country_id]));
 
       free(name);
 

--- a/src/init.cc
+++ b/src/init.cc
@@ -31,13 +31,13 @@ extern "C" {
     Org::Init(exports);
 
     // Utility memeber method
-    Local<FunctionTemplate> check = NanNewLocal<FunctionTemplate>(FunctionTemplate::New(utils::check));
-    Local<FunctionTemplate> isString = NanNewLocal<FunctionTemplate>(FunctionTemplate::New(utils::isString));
-    exports->Set(String::NewSymbol("check"), check->GetFunction());
-    exports->Set(String::NewSymbol("isString"), isString->GetFunction());
+    Local<FunctionTemplate> check = NanNew<FunctionTemplate>(utils::check);
+    Local<FunctionTemplate> isString = NanNew<FunctionTemplate>(utils::isString);
+    exports->Set(NanSymbol("check"), check->GetFunction());
+    exports->Set(NanSymbol("isString"), isString->GetFunction());
 
     // Meta infomation
-    exports->Set(String::NewSymbol("libgeoip"), String::New(GeoIP_lib_version()));
+    exports->Set(NanSymbol("libgeoip"), NanNew<String>(GeoIP_lib_version()));
   }
 
   NODE_MODULE(native, InitAll)

--- a/src/netspeed.cc
+++ b/src/netspeed.cc
@@ -22,14 +22,14 @@ Persistent<FunctionTemplate> NetSpeed::constructor_template;
 void NetSpeed::Init(Handle<Object> exports) {
     NanScope();
 
-    Local<FunctionTemplate> tpl = NanNewLocal<FunctionTemplate>(FunctionTemplate::New(New));
-    NanAssignPersistent(FunctionTemplate, constructor_template, tpl);
+    Local<FunctionTemplate> tpl = NanNew<FunctionTemplate>(New);
+    NanAssignPersistent(constructor_template, tpl);
     tpl->InstanceTemplate()->SetInternalFieldCount(1);
-    tpl->SetClassName(String::NewSymbol("NetSpeed"));
+    tpl->SetClassName(NanSymbol("NetSpeed"));
 
-    tpl->PrototypeTemplate()->Set(String::NewSymbol("lookupSync"),
-            FunctionTemplate::New(lookupSync)->GetFunction());
-    exports->Set(String::NewSymbol("NetSpeed"), tpl->GetFunction());
+    tpl->PrototypeTemplate()->Set(NanSymbol("lookupSync"),
+            NanNew<FunctionTemplate>(lookupSync)->GetFunction());
+    exports->Set(NanSymbol("NetSpeed"), tpl->GetFunction());
 }
 
 NAN_METHOD(NetSpeed::New) {
@@ -64,29 +64,31 @@ NAN_METHOD(NetSpeed::lookupSync) {
 
     NetSpeed *n = ObjectWrap::Unwrap<NetSpeed>(args.This());
 
-    Local<Value> data = NanNewLocal<Value>(Null());
-    Local<String> host_str = NanNewLocal<String>(args[0]->ToString());
-    char host_cstr[host_str->Length() + 1];
-    NanFromV8String(args[0].As<Object>(), Nan::ASCII, NULL, host_cstr, host_str->Length() + 1, v8::String::HINT_MANY_WRITES_EXPECTED);
+    Local<Value> data = NanNew(NanNull());
+    Local<String> host_str = args[0]->ToString();
+    size_t size = host_str->Length() + 1;
+    char host_cstr[size];
+    size_t bc;
+    NanCString(args[0], &bc, host_cstr, size);
 
     uint32_t ipnum = _GeoIP_lookupaddress(host_cstr);
 
     if (ipnum <= 0) {
-        NanReturnValue(Null());
+        NanReturnValue(NanNull());
     }
 
     int netspeed = GeoIP_id_by_ipnum(n->db, ipnum);
 
     if (netspeed < 0) {
-        NanReturnValue(Null());
+        NanReturnValue(NanNull());
     } else if (netspeed == GEOIP_UNKNOWN_SPEED) {
-        data = String::New("Unknown");
+        data = NanNew<String>("Unknown");
     } else if (netspeed == GEOIP_DIALUP_SPEED) {
-        data = String::New("Dialup");
+        data = NanNew<String>("Dialup");
     } else if (netspeed == GEOIP_CABLEDSL_SPEED) {
-        data = String::New("CableDSL");
+        data = NanNew<String>("CableDSL");
     } else if (netspeed == GEOIP_CORPORATE_SPEED) {
-        data = String::New("Corporate");
+        data = NanNew<String>("Corporate");
     }
 
     NanReturnValue(data);

--- a/src/netspeedcell.cc
+++ b/src/netspeedcell.cc
@@ -22,14 +22,14 @@ Persistent<FunctionTemplate> NetSpeedCell::constructor_template;
 void NetSpeedCell::Init(Handle<Object> exports) {
  NanScope();
 
-  Local<FunctionTemplate> tpl = NanNewLocal<FunctionTemplate>(FunctionTemplate::New(New));
-  NanAssignPersistent(FunctionTemplate, constructor_template, tpl);
+  Local<FunctionTemplate> tpl = NanNew<FunctionTemplate>(New);
+  NanAssignPersistent(constructor_template, tpl);
   tpl->InstanceTemplate()->SetInternalFieldCount(1);
-  tpl->SetClassName(String::NewSymbol("NetSpeedCell"));
+  tpl->SetClassName(NanSymbol("NetSpeedCell"));
 
-  tpl->PrototypeTemplate()->Set(String::NewSymbol("lookupSync"),
-      FunctionTemplate::New(lookupSync)->GetFunction());
-  exports->Set(String::NewSymbol("NetSpeedCell"), tpl->GetFunction());
+  tpl->PrototypeTemplate()->Set(NanSymbol("lookupSync"),
+      NanNew<FunctionTemplate>(lookupSync)->GetFunction());
+  exports->Set(NanSymbol("NetSpeedCell"), tpl->GetFunction());
 }
 
 NAN_METHOD(NetSpeedCell::New) {
@@ -62,17 +62,19 @@ NAN_METHOD(NetSpeedCell::lookupSync) {
 
   NetSpeedCell *n = ObjectWrap::Unwrap<NetSpeedCell>(args.This());
 
-  Local<Value> data = NanNewLocal<Value>(Null());
-  Local<String> host_str = NanNewLocal<String>(args[0]->ToString());
-  char host_cstr[host_str->Length() + 1];
-  NanFromV8String(args[0].As<Object>(), Nan::ASCII, NULL, host_cstr, host_str->Length() + 1, v8::String::HINT_MANY_WRITES_EXPECTED);
+  Local<Value> data = NanNew(NanNull());
+  Local<String> host_str = args[0]->ToString();
+  size_t size = host_str->Length() + 1;
+  char host_cstr[size];
+  size_t bc;
+  NanCString(args[0], &bc, host_cstr, size);
 
   char *speed = GeoIP_name_by_addr(n->db, host_cstr);
   if (!speed) {
-    data = String::New("Unknown");
+    data = NanNew<String>("Unknown");
   }
   else {
-    data = String::New(speed);
+    data = NanNew<String>(speed);
   }
 
   NanReturnValue(data);

--- a/src/org.cc
+++ b/src/org.cc
@@ -21,14 +21,14 @@ Persistent<FunctionTemplate> Org::constructor_template;
 void Org::Init(Handle<Object> exports) {
   NanScope();
 
-  Local<FunctionTemplate> tpl = NanNewLocal<FunctionTemplate>(FunctionTemplate::New(New));
-  NanAssignPersistent(FunctionTemplate, constructor_template, tpl);
+  Local<FunctionTemplate> tpl = NanNew<FunctionTemplate>(New);
+  NanAssignPersistent(constructor_template, tpl);
   tpl->InstanceTemplate()->SetInternalFieldCount(1);
-  tpl->SetClassName(String::NewSymbol("Org"));
+  tpl->SetClassName(NanSymbol("Org"));
 
-  tpl->PrototypeTemplate()->Set(String::NewSymbol("lookupSync"),
-      FunctionTemplate::New(lookupSync)->GetFunction());
-  exports->Set(String::NewSymbol("Org"), tpl->GetFunction());
+  tpl->PrototypeTemplate()->Set(NanSymbol("lookupSync"),
+      NanNew<FunctionTemplate>(lookupSync)->GetFunction());
+  exports->Set(NanSymbol("Org"), tpl->GetFunction());
 }
 
 NAN_METHOD(Org::New) {
@@ -62,26 +62,28 @@ NAN_METHOD(Org::New) {
 NAN_METHOD(Org::lookupSync) {
   NanScope();
 
-  Local<Value> data = NanNewLocal<Value>(Null());
-  Local<String> host_str = NanNewLocal<String>(args[0]->ToString());
-  char host_cstr[host_str->Length() + 1];
-  NanFromV8String(args[0].As<Object>(), Nan::ASCII, NULL, host_cstr, host_str->Length() + 1, v8::String::HINT_MANY_WRITES_EXPECTED);
+  Local<Value> data = NanNew(NanNull());
+  Local<String> host_str = args[0]->ToString();
+  size_t size = host_str->Length() + 1;
+  char host_cstr[size];
+  size_t bc;
+  NanCString(args[0], &bc, host_cstr, size);
   Org *o = ObjectWrap::Unwrap<Org>(args.This());
 
   uint32_t ipnum = _GeoIP_lookupaddress(host_cstr);
 
   if (ipnum <= 0) {
-    NanReturnValue(Null());
+    NanReturnValue(NanNull());
   }
 
   char *org = GeoIP_org_by_ipnum(o->db, ipnum);
   if (!org) {
-    NanReturnValue(Null());
+    NanReturnValue(NanNull());
   }
 
   char *name = _GeoIP_iso_8859_1__utf8(org);
 
-  data = String::New(name);
+  data = NanNew<String>(name);
 
   free(org);
   free(name);

--- a/src/region.cc
+++ b/src/region.cc
@@ -23,14 +23,14 @@ Persistent<FunctionTemplate> Region::constructor_template;
 void Region::Init(Handle<Object> exports) {
   NanScope();
 
-  Local<FunctionTemplate> tpl = NanNewLocal<FunctionTemplate>(FunctionTemplate::New(New));
-  NanAssignPersistent(FunctionTemplate, constructor_template, tpl);
+  Local<FunctionTemplate> tpl = NanNew<FunctionTemplate>(New);
+  NanAssignPersistent(constructor_template, tpl);
   tpl->InstanceTemplate()->SetInternalFieldCount(1);
-  tpl->SetClassName(String::NewSymbol("Region"));
+  tpl->SetClassName(NanSymbol("Region"));
 
-  tpl->PrototypeTemplate()->Set(String::NewSymbol("lookupSync"),
-      FunctionTemplate::New(lookupSync)->GetFunction());
-  exports->Set(String::NewSymbol("Region"), tpl->GetFunction()); 
+  tpl->PrototypeTemplate()->Set(NanSymbol("lookupSync"),
+      NanNew<FunctionTemplate>(lookupSync)->GetFunction());
+  exports->Set(NanSymbol("Region"), tpl->GetFunction());
 }
 
 NAN_METHOD(Region::New) {
@@ -63,23 +63,25 @@ NAN_METHOD(Region::New) {
 NAN_METHOD(Region::lookupSync) {
   NanScope();
 
-  Local<Object> data = NanNewLocal<Object>(Object::New());
-  Local<String> host_str = NanNewLocal<String>(args[0]->ToString());
-  char host_cstr[host_str->Length() + 1];
-  NanFromV8String(args[0].As<Object>(), Nan::ASCII, NULL, host_cstr, host_str->Length() + 1, v8::String::HINT_MANY_WRITES_EXPECTED);
+  Local<Object> data = NanNew<Object>();
+  Local<String> host_str = args[0]->ToString();
+  size_t size = host_str->Length() + 1;
+  char host_cstr[size];
+  size_t bc;
+  NanCString(args[0], &bc, host_cstr, size);
   Region *r = ObjectWrap::Unwrap<Region>(args.This());
 
   uint32_t ipnum = _GeoIP_lookupaddress(host_cstr);
 
   if (ipnum <= 0) {
-    NanReturnValue(Null());
+    NanReturnValue(NanNull());
   }
 
   GeoIPRegion *region = GeoIP_region_by_ipnum(r->db, ipnum);
 
   if (region) {
-    data->Set(String::NewSymbol("country_code"), String::New(region->country_code));
-    data->Set(String::NewSymbol("region"), String::New(region->region));
+    data->Set(NanSymbol("country_code"), NanNew<String>(region->country_code));
+    data->Set(NanSymbol("region"), NanNew<String>(region->region));
     GeoIPRegion_delete(region);
     NanReturnValue(data);
   } else {

--- a/src/utils.cc
+++ b/src/utils.cc
@@ -12,7 +12,7 @@ namespace native {
         NAN_METHOD(isString) {
             NanScope();
 
-            Local<Boolean> result = NanNewLocal<Boolean>(args[0].As<Object>()->IsString() ? True() : False());
+            Local<Boolean> result = NanNew(args[0].As<Object>()->IsString() ? NanTrue() : NanFalse());
 
             NanReturnValue(result);
         }
@@ -20,10 +20,12 @@ namespace native {
         NAN_METHOD(check) {
             NanScope();
 
-            Local<Value> edition = NanNewLocal<Value>(Null());
-            Local<String> file_str = NanNewLocal<String>(args[0]->ToString());
-            char file_cstr[file_str->Length() + 1];
-            NanFromV8String(args[0].As<Object>(), Nan::ASCII, NULL, file_cstr, file_str->Length() + 1, v8::String::HINT_MANY_WRITES_EXPECTED);
+            Local<Value> edition = NanNew(NanNull());
+            Local<String> file_str = args[0]->ToString();
+            size_t size = file_str->Length() + 1;
+            char file_cstr[size];
+            size_t bc;
+            NanCString(args[0], &bc, file_cstr, size);
 
             GeoIP *db = GeoIP_open(file_cstr, GEOIP_STANDARD);
 
@@ -32,67 +34,67 @@ namespace native {
 
                 switch(db_edition) {
                     case GEOIP_COUNTRY_EDITION:
-                        edition = String::New("country");
+                        edition = NanNew<String>("country");
                         break;
 
                     case GEOIP_COUNTRY_EDITION_V6:
-                        edition = String::New("country_v6");
+                        edition = NanNew<String>("country_v6");
                         break;
 
                     case GEOIP_CITY_EDITION_REV0:
-                        edition = String::New("city");
+                        edition = NanNew<String>("city");
                         break;
 
                     case GEOIP_CITY_EDITION_REV1:
-                        edition = String::New("city");
+                        edition = NanNew<String>("city");
                         break;
 
                     case GEOIP_CITY_EDITION_REV0_V6:
-                        edition = String::New("city_v6");
+                        edition = NanNew<String>("city_v6");
                         break;
 
                     case GEOIP_CITY_EDITION_REV1_V6:
-                        edition = String::New("city_v6");
+                        edition = NanNew<String>("city_v6");
                         break;
 
                     case GEOIP_REGION_EDITION_REV0:
-                        edition = String::New("region");
+                        edition = NanNew<String>("region");
                         break;
 
                     case GEOIP_REGION_EDITION_REV1:
-                        edition = String::New("region");
+                        edition = NanNew<String>("region");
                         break;
 
                     case GEOIP_ORG_EDITION:
-                        edition = String::New("org");
+                        edition = NanNew<String>("org");
                         break;
 
                     case GEOIP_ISP_EDITION:
-                        edition = String::New("isp");
+                        edition = NanNew<String>("isp");
                         break;
 
                     case GEOIP_ASNUM_EDITION:
-                        edition = String::New("asnum");
+                        edition = NanNew<String>("asnum");
                         break;
 
                     case GEOIP_PROXY_EDITION:
-                        edition = String::New("proxy");
+                        edition = NanNew<String>("proxy");
                         break;
 
                     case GEOIP_NETSPEED_EDITION:
-                        edition = String::New("netspeed");
+                        edition = NanNew<String>("netspeed");
                         break;
 
                     case GEOIP_NETSPEED_EDITION_REV1:
-                        edition = String::New("netspeed cellular");
+                        edition = NanNew<String>("netspeed cellular");
                         break;
 
                     case GEOIP_DOMAIN_EDITION:
-                        edition = String::New("domain");
+                        edition = NanNew<String>("domain");
                         break;
 
                     default:
-                        edition = String::New("unknown");
+                        edition = NanNew<String>("unknown");
                 }
 
                 GeoIP_delete(db);

